src/modules/commander/HomePosition.cpp

```diff
@@ -319,12 +336,51 @@ void HomePosition::update(bool set_automatically, bool check_if_changed)
 		_gps_epv = vehicle_gps_position.epv;
 
 		const hrt_abstime now = hrt_absolute_time();
-		const bool time = (now < vehicle_gps_position.timestamp + 1_s);
-		const bool fix = vehicle_gps_position.fix_type >= kHomePositionGPSRequiredFixType;
-		const bool eph = vehicle_gps_position.eph < kHomePositionGPSRequiredEPH;
-		const bool epv = vehicle_gps_position.epv < kHomePositionGPSRequiredEPV;
-		const bool evh = vehicle_gps_position.s_variance_m_s < kHomePositionGPSRequiredEVH;
-		_gps_position_for_home_valid = time && fix && eph && epv && evh;
+		const bool time_valid = now < (vehicle_gps_position.timestamp + 1_s);
+		const bool fix_valid = vehicle_gps_position.fix_type >= kHomePositionGPSRequiredFixType;
+		const bool eph_valid = vehicle_gps_position.eph < kHomePositionGPSRequiredEPH;
+		const bool epv_valid = vehicle_gps_position.epv < kHomePositionGPSRequiredEPV;
+		const bool evh_valid = vehicle_gps_position.s_variance_m_s < kHomePositionGPSRequiredEVH;
+
+		_gps_position_for_home_valid = time_valid && fix_valid && eph_valid && epv_valid && evh_valid;
+
+		if (_gps_position_for_home_valid && _last_gps_timestamp != 0 && _last_baro_timestamp != 0) {
+
+			const float gps_alt = static_cast<float>(_gps_alt);
+
+			if (!_gps_vel_integral_init) {
```

Comment:
You could also init _gps_vel_integral with NAN and then do 
`if (!PX4_ISFINITE(_gps_vel_integral))` here